xcvm/architecture/transports/otp.md

```diff
@@ -0,0 +1,208 @@
+# Open Transport Protocol (OTP)
+
+Although [IBC](./ibc.md) is the default and preferred transport method within the `XCVM`, we also support arbitrary third party protocols. The abstract interface for registration and participation in `XCVM` transactions is what we call `OTP`.
+
+For a transport to be OTP compatible, it has to support the following operations:
+
+1. Fungible token transfers.
+2. Opaque data transfers.
+
+Ideally, a protocol supports both in a single operation, although most bridges will need to be wrapped in an additional set of contracts to be fully compatible.
+
+`OTP` handles the following operations:
+
+1. Maintaining a blacklist of transports.
+2. Ingress of messages on the chain.
+3. Registration of routing.
+4. Egress of messages on the chain. 
+
+## Blacklist
+
+The blacklist is used to enact emergency operations in case of bridge fraud, which can be caused by hacks, or bugs within the system. The blacklist globally disables bridges based on `LAYR` governance.
+
+Note that the blacklist is not used to manage ecosystem specific risk, such as toxic assets. It only concerns itself with bridges where the message authenticity can no longer be guaranteed.
+
+## XCVM Ingress
+
+A transport does not need to register itself to pass messages into the `XCVM`, it can do so directly, assuming that the interpreter instance accepting the transport's messages has whitelisted this transport. 
```

Comment:
not clear what it means. is that chain which is initiator of XCVM, creates XCVM instance with whilelisted set of transports?  and then that list transferred to XCVM on receiving side chain?

Reply:
The receiving side needs to have it whitelisted first, so if a contract is automatically setting up instances, it would do so in the following way:

1. Create instructions and send to other chain over IBC
2. On other chain, execute instructions. One of these instructions should contain Call(XcvmInterpreter::SetFilter(address)), where address is a contract that has signature: fn(bridge, identity, program) -> result<bool, str>. If it is unset, the xcvm interpreter will check that bridge == ibc. If set to something custom, the 3rd party dev can do it based on the 3 inputs, e.g. accept programs over wormhole if the program does not contain Transfer.